pkg/apis/rollouts/v1alpha1/types.go

```diff
@@ -342,6 +342,8 @@ type IstioTrafficRouting struct {
 type IstioVirtualService struct {
 	// Name holds the name of the VirtualService
 	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+	// Namespace holds the namespace of the VirtualService
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
Since the VirtualService name needs to be specified like `stable-svc.<namespace>` when referencing across namespaces, is there any value in introducing a `Namespace` field to the VirtualService reference?

Can we just infer namespace instead? This eliminates the possibility of the two values disagreeing with each other due to user error.

Reply:
I added the `namespace` field just because it is easy to process in the code.
After second thought, I agreed that inferring namespace in `stable-svc.<namespace>` is better liking inferring service endpoint. I will make the change. Thanks.

Reply:
@jessesuen , PR is updated. Please take a another look at it. Thanks.